.github/workflows/CI_build.yml

```
@@ -7,9 +7,79 @@ on:
 
 jobs:
 
-  analyze:
-    name: analyze / python-${{ matrix.python-version }} / ${{ matrix.os }}
+  first_check:
+    name: first code check / python-3.8 / ubuntu-latest
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.8  
+      - name: Python info
+        run: |
+          which python
+          python --version
+      - name: Build package and create dev environment
+        run: |
+          python -m pip install --upgrade pip
+          python setup.py build
+          pip install -e .[dev]
+      - name: Test with coverage
+        run: |
+          pytest --cov --cov-report term --cov-report xml --junitxml=xunit-result.xml
+      - name: Correct coverage paths
+        run: sed -i "s+$PWD/++g" coverage.xml
+      - name: Check style against standards using prospector
+        shell: bash -l {0}
+        run: prospector -o grouped -o pylint:pylint-report.txt
+      - name: Check whether import statements are used consistently
+        shell: bash -l {0}
+        run: isort --check-only --diff --conda-env spec2vec-dev --recursive --wrap-length 79 --lines-after-imports 2 --force-single-line --no-lines-before FUTURE --no-lines-before STDLIB --no-lines-before THIRDPARTY --no-lines-before FIRSTPARTY --no-lines-before LOCALFOLDER spec2vec/ tests/ integration-tests/
+          
+  build_pypi:
+    name: Pypi build / python-${{ matrix.python-version }} / ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    needs: first_check
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
+        python-version: ['3.7', '3.8']
```

You could exclude `ubuntu-latest`  + `py3.8` here with
```suggestion
        python-version: ['3.7', '3.8']
        exclude:
           # already tested in first_check job
           - python-version: 3.8
             os: ubuntu-latest
```

---

Done.